org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/CodeActionMarkerResolution.java

```diff
@@ -55,6 +64,18 @@ public String getLabel() {
 
 	@Override
 	public void run(IMarker marker) {
+		if (codeAction.getEdit() == null) {
+			String languageServerId = marker.getAttribute(LSPDiagnosticsToMarkers.LANGUAGE_SERVER_ID, (String)null);
+			LanguageServerDefinition definition = languageServerId != null ? LanguageServersRegistry.getInstance().getDefinition(languageServerId) : null;
```

Comment:
I think the resolve code action must be done with the same languageserver instance which have created the codeaction.
I will set this instance in the CodeActionMarkerResolution constructor.

The another thing which is important is to check if server has the capability to resolve codeaction.

Reply:
Yes, I agree with both open points, they should be done.

Reply:
> I think the resolve code action must be done with the same languageserver instance which have created the codeaction. I will set this instance in the CodeActionMarkerResolution constructor.

OK

> The another thing which is important is to check if server has the capability to resolve codeaction.

OK